libsolutil/CommonData.h

```diff
@@ -450,6 +450,32 @@ bool contains_if(T const& _t, Predicate const& _p)
 	return std::end(_t) != std::find_if(std::begin(_t), std::end(_t), _p);
 }
 
+/// Function that insert @param _elem into a vector @param _destination only if _elem is not
+/// already present in _destination vector.
+template <class T>
+void emplaceBackUnique(std::vector<T>& _destination, T _elem)
+{
+	if (!contains(_destination, _elem))
+		_destination.emplace_back(std::move(_elem));
+}
+
+/// Function that takes vector @param _destination and appends vector @param _source,
+/// whilst only taking into account unique elements of _source.
+template <class T>
+void concatenateVectorWithoutDuplicates(std::vector<T>& _destination, std::vector<T> const& _source)
+{
+	for (auto&& elem: _source)
+		emplaceBackUnique(_destination, elem);
+}
+
+/// Function that removes all elements present in vector @param _sub from vector @param _super.
+template <class T>
+void removeVectorSubset(std::vector<T>& _super, std::vector<T> const& _sub)
+{
+	for (auto const& elem: _sub)
+		_super.erase(std::find(_super.begin(), _super.end(), elem));
+}
```

Comment:
I think naming could be clearer here (e.g. I'd use `tryEmplaceBack()` analogous to `map::try_emplace()` or `appendMissing()` since `concatenate...()` sounds like it would be symmetrical and remove duplicates from both vectors, which it doesn't).

But before getting into that, maybe first say what do you think about my suggestion above to use a custom comparator, since it would make these helpers unnecessary in the first place.

Reply:
I picked shit names on purpose because I was 100% certain you'd swoop in with better ones :)